src/plugins/csv/csvplugin.cpp

```diff
@@ -43,62 +45,64 @@ CsvPlugin::CsvPlugin()
 
 bool CsvPlugin::write(const Map *map, const QString &fileName)
 {
+    // Extract file name without extension and path
+    QFileInfo fileInfo(fileName);
+    fileInfo.setCaching(false);
+    const QString fileNameWoExtension = fileInfo.baseName();
```

Comment:
I'd replace `baseName` with `completeBaseName` to avoid cutting off more than just the `.csv` part in the case of multiple dots.